src/tendencies/advection.jl

```diff
@@ -77,20 +77,20 @@ function explicit_vertical_advection_tendency!(Yₜ, Y, p, t, colidx)
     # Mass conservation
     ᶜJ = Fields.local_geometry_field(axes(ᶜρ)).J
 
-
     # Momentum conservation
     @. ᶠω¹²[colidx] += ᶠcurlᵥ(ᶜuₕ[colidx])
     @. ᶠu¹²[colidx] =
-        Geometry.project(Geometry.Contravariant12Axis(), ᶠinterp(ᶜuvw[colidx]))
-    @. ᶠu³[colidx] = Geometry.project(
-        Geometry.Contravariant3Axis(),
-        C123(ᶠinterp(ᶜuₕ[colidx])) + C123(ᶠw[colidx]),
-    )
+        Geometry.project(Geometry.Contravariant12Axis(), ᶠuvw[colidx])
+    @. ᶠu³[colidx] =
+        Geometry.project(Geometry.Contravariant3Axis(), ᶠuvw[colidx])
     @. Yₜ.c.uₕ[colidx] -=
-        ᶜinterp(ᶠω¹²[colidx] × ᶠu³[colidx]) +
+        ᶜinterp(ᶠω¹²[colidx] × ᶠinterp(ᶜρ *ᶜ J) * ᶠu³[colidx]) / (ᶜρ * ᶜJ) +
```

Comment:
Given that we have pre-computed \tilde{u}, we can then implement equation 26 in the design docs here, so that the contravariant component is interpolated